### PR TITLE
Update scrape-config for istio-ingress.

### PR DESCRIPTION
**Pull request by axel7born**
* Update scrape config to get metrics from istio-ingress pods in other namespaces.
* Adapt istio-ingress dashboard to show metrics summarized per namespace.

**How to categorize this PR?**

/area networking monitoring
/kind bug

**What this PR does / why we need it**:
In the Grafana dashboard `Istio Ingress Gateway` only metrics from pod in the default namespace `istio-ingress` were visible. However, with the `ExposureClasses` and the introduction of Highly Available Shoot Control Planes, ingress-gateway pod are also deployed in other namespaces. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```fix operator
Fix missing metrics from istio ingress-gateways.
```

### CODE REVIEW

File: pkg/operation/botanist/component/istio/monitoring.go [hunk 1]
```diff
@@ -170,15 +170,15 @@ metric_relabel_configs:
 metrics_path: /stats/prometheus
 kubernetes_sd_configs:
 - role: endpoints
-  namespaces:
-    names: [ ` + v1beta1constants.DefaultSNIIngressNamespace + ` ]
+  selectors:
+   label: istio=ingressgateway
```
Comment (ScheererJ) [inline pkg/operation/botanist/component/istio/monitoring.go]: Shouldn't it be two space indentation?

```suggestion
    label: istio=ingressgateway
```

File: pkg/operation/botanist/component/istio/monitoring.go [hunk 2]
```diff
@@ -170,15 +170,15 @@ metric_relabel_configs:
...
 relabel_configs:
 - source_labels:
   - __meta_kubernetes_service_name
   - __meta_kubernetes_endpoint_port_name
   - __meta_kubernetes_namespace
   action: keep
-  regex: ` + v1beta1constants.DefaultSNIIngressServiceName + `;tls-tunnel;` + v1beta1constants.DefaultSNIIngressNamespace + `
+  regex: ` + v1beta1constants.DefaultSNIIngressServiceName + `;tls-tunnel;` + v1beta1constants.DefaultSNIIngressNamespace + `.*
```
Comment (ScheererJ) [inline pkg/operation/botanist/component/istio/monitoring.go]: What if https://github.com/gardener/gardener/blob/b2d32fe231cb47926480eb30a1d8f2ca6b9be4dc/example/20-componentconfig-gardenlet.yaml#L153 is changed?
What if https://github.com/gardener/gardener/blob/b2d32fe231cb47926480eb30a1d8f2ca6b9be4dc/example/20-componentconfig-gardenlet.yaml#L168 is changed?

**Review by ScheererJ — CHANGES_REQUESTED** [review]

File: pkg/operation/botanist/component/istio/monitoring.go (hunk 2 above)
Comment (axel7born) [reply pkg/operation/botanist/component/istio/monitoring.go]: I removed the namespace from the regex.

File: pkg/operation/botanist/component/istio/monitoring.go (hunk 1 above)
Comment (axel7born) [reply pkg/operation/botanist/component/istio/monitoring.go]: fixed

**Comment by axel7born** [comment]
/cc @istvanballok @rickardsjp 
Could you take a look, if the changes in the scrape config make sense?

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Comment by ScheererJ** [comment]
/hold
waiting for feedback from @istvanballok / @rickardsjp

File: charts/seed-bootstrap/dashboards/istio/istio-ingress-gateway-dashboard.json [hunk 3]
```diff
@@ -208,12 +314,116 @@
...
+      "steppedLine": false,
+      "targets": [
+        {
+          "exemplar": true,
+          "expr": "sum(rate(container_memory_working_set_bytes{pod=~\"$istio_proxy_pod\"}[$__rate_interval])) by (namespace)",
+          "format": "time_series",
+          "hide": false,
+          "interval": "",
+          "intervalFactor": 2,
+          "legendFormat": "{{ pod }}",
```
Comment (istvanballok) [inline charts/seed-bootstrap/dashboards/istio/istio-ingress-gateway-dashboard.json]:
```suggestion
          "legendFormat": "{{ namespace }}",
```

I think there is a similar typo in the panel above but I could not annotate it in the review as the respective line did not change.

Note that Grafana seems to do the "right" thing if the result of the query has a single label, and even if it does not match the label expression in the legend configuration, the label values are shown. Nevertheless, we could still adjust the Legend expression for consistency.

image

**Review by istvanballok — APPROVED** [review]
/lgtm with a minor comment

File: charts/seed-bootstrap/dashboards/istio/istio-ingress-gateway-dashboard.json (hunk 3 above)
Comment (axel7born) [reply charts/seed-bootstrap/dashboards/istio/istio-ingress-gateway-dashboard.json]: Fixed.

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Pull request merged**